src/connectedk8s/azext_connectedk8s/_precheckutils.py

```diff
@@ -0,0 +1,219 @@
+# --------------------------------------------------------------------------------------------
+
+import os
+import shutil
+import subprocess
+from subprocess import Popen, PIPE
+import time
+import requests
+from requests.adapters import HTTPAdapter
+from urllib3.util.retry import Retry
+import json
+from kubernetes import client, config, watch, utils
+from knack.util import CLIError
+from knack.log import get_logger
+from knack.prompting import NoTTYException, prompt_y_n
+from azure.cli.core.commands.client_factory import get_subscription_id
+from azure.cli.core.util import send_raw_request
+from azure.cli.core import telemetry
+from azure.core.exceptions import ResourceNotFoundError, HttpResponseError
+from msrest.exceptions import AuthenticationError, HttpOperationError, TokenExpiredError
+from msrest.exceptions import ValidationError as MSRestValidationError
+from kubernetes.client.rest import ApiException
+from azext_connectedk8s._client_factory import _resource_client_factory, _resource_providers_client
+import azext_connectedk8s._constants as consts
+import azext_connectedk8s._utils as azext_utils
+from kubernetes import client as kube_client
+from azure.cli.core import get_default_cli
+from azure.cli.core.azclierror import CLIInternalError, ClientRequestError, ArgumentUsageError, ManualInterrupt, AzureResponseError, AzureInternalError, ValidationError
+from argparse import Namespace
+from pydoc import cli
+from logging import exception
+import yaml
+import json
+import datetime
+from subprocess import Popen, PIPE, run, STDOUT, call, DEVNULL
+import shutil
+from knack.log import get_logger
+from azure.cli.core import telemetry
+import azext_connectedk8s._constants as consts
+logger = get_logger(__name__)
+# pylint: disable=unused-argument, too-many-locals, too-many-branches, too-many-statements, line-too-long
+# pylint: disable
+
+
+def check_preonboarding_inspector_container(corev1_api_instance, batchv1_api_instance, absolute_path, helm_client_location, kubectl_client_location, kube_config, kube_context, http_proxy, https_proxy, no_proxy, proxy_cert):
+    try:
+        # Setting DNS and Outbound Check as working
+        dns_check = "Starting"
+        outbound_connectivity_check = "Starting"
+        # Executing the pre onboarding inspector job and fetching the logs obtained
+        preonboarding_inspector_container_log = executing_preonboarding_inspector_job(corev1_api_instance, batchv1_api_instance, absolute_path, helm_client_location, kubectl_client_location, kube_config, kube_context, http_proxy, https_proxy, no_proxy, proxy_cert)
+        # If preonboarding_inspector_container_log is not empty then only we will check for the results
+        if(preonboarding_inspector_container_log is not None and preonboarding_inspector_container_log != ""):
+            preonboarding_inspector_container_log_list = preonboarding_inspector_container_log.split("\n")
```

Comment:
Line 57-69 we can make it common. Create a new method named parse_container_log_for_result() and add this common code there. Let's call the method from both diagnoser and pre-onboarding inspector